common/matd.c

```diff
@@ -51,9 +51,10 @@ matd_t *matd_create(int rows, int cols)
     if (rows == 0 || cols == 0)
         return matd_create_scalar(0);
 
-    matd_t *m = calloc(1, sizeof(matd_t) + (rows*cols*sizeof(double)));
+    matd_t *m = calloc(1, sizeof(matd_t));
     m->nrows = rows;
     m->ncols = cols;
+    m->data = calloc(rows * cols, sizeof(TYPE));
```

Comment:
This has the potential to leek memory, if `TYPE` gets redefined. We should either stick with `double` wherever `matd_t` is used, or replace all usages of `double` with `matd_t` by `TYPE` macro.